src/main/java/uk/gov/hmcts/darts/usermanagement/service/impl/SecurityGroupServiceImpl.java

```diff
@@ -82,6 +96,87 @@ public List<SecurityGroupWithIdAndRole> getSecurityGroups(List<Integer> roleIds,
             .map(securityGroupCourthouseMapper::mapToSecurityGroupWithIdAndRoleWithCourthouse).toList();
     }
 
+    @Transactional
+    @Override
+    public SecurityGroupWithIdAndRoleAndUsers modifySecurityGroup(Integer securityGroupId, SecurityGroupPatch securityGroupPatch) {
+
+        Optional<SecurityGroupEntity> securityGroupEntityOptional = securityGroupRepository.findById(securityGroupId);
+
+        if (securityGroupEntityOptional.isPresent()) {
+            SecurityGroupEntity securityGroupEntity = securityGroupEntityOptional.get();
+            updateSecurityGroupEntity(securityGroupPatch, securityGroupEntity);
+            securityGroupRepository.saveAndFlush(securityGroupEntity);
+            return securityGroupCourthouseMapper.mapToSecurityGroupWithCourthousesAndUsers(securityGroupEntity);
+        } else {
+            //throw a 404 not found
+            throw new DartsApiException(
+                UserManagementError.SECURITY_GROUP_NOT_FOUND,
+                String.format("Security group id %d not found", securityGroupId));
+        }
+
+    }
+
+    protected void updateSecurityGroupEntity(SecurityGroupPatch securityGroupPatch, SecurityGroupEntity securityGroupEntity) {
+        validate(securityGroupPatch, securityGroupEntity);
+
+        String name = securityGroupPatch.getName();
+        String displayName = securityGroupPatch.getDisplayName();
+
+        if (name != null) {
+            securityGroupEntity.setGroupName(name);
+        }
+        if (displayName != null) {
+            securityGroupEntity.setDisplayName(displayName);
+        }
+        String description = securityGroupPatch.getDescription();
+        if (description != null) {
+            securityGroupEntity.setDescription(description);
+        }
+        List<Integer> courthouseIds = securityGroupPatch.getCourthouseIds();
+        if (courthouseIds != null) {
+            Set<CourthouseEntity> courthouseEntities = new HashSet<>();
+            for (Integer courthouseId: courthouseIds) {
+                Optional<CourthouseEntity> courthouseEntity = courthouseRepository.findById(courthouseId);
+                if (courthouseEntity.isPresent()) {
+                    courthouseEntities.add(courthouseEntity.get());
+                } else {
+                    throw new DartsApiException(
+                        UserManagementError.COURTHOUSE_NOT_FOUND,
+                        String.format("Courthouse id %d not found", courthouseId));
+                }
+            }
+            securityGroupEntity.setCourthouseEntities(courthouseEntities);
```

Comment:
Looking at the ManyToMany association here, we seem to have another join table definition `CourthouseEntity#securityGroups`

I think it should just be `@ManyToMany(mappedBy = SecurityGroupEntity_.COURTHOUSE_ENTITIES)`

Since the association is bi-directional, I think we are supposed to manage the other side of the relationship too, so that the entity manager knows about the opposite way back to security_group from courthouse?

So wondering if we might get unexpected results somewhere else, depending on how the courthouse service / management is used perhaps - something to look out for during testing